### ISSUE TITLE
FE "Other nearby Vet Centers" list on Vet Center location page isn't honoring distance correctly

### ISSUE THREAD

**Issue by dsinla** [department-of-veterans-affairs/va.gov-cms#9924]
## Describe the defect
The "Other nearby Vet Centers" list on the Vet Center location page doesn't list locations in order of distance. 

## To Reproduce

Originally reported by Tiffany.Lever@va.gov

1. Navigate to this page: http://preview-prod.vfs.va.gov/preview?nodeId=28177
2. View the "Other nearby Vet Centers list

Note the 4 listed:

Fayetteville, NC Vet Center                115-123 miles from MB VC driving distance  (95.57 ATCF)
Charleston, SC Vet Center                99-104 miles from MB VC driving distance (83.98 ATCF)                            
Jacksonville, NC Vet Center              138-144 miles from MB VC (116 ATCF)
Charlotte Vet Center                         188-191 miles from MB VC (154.95 ATCF)  

Note that Charleston is closer than Fayetteville, yet it is listed 2nd
Also, the Colombia, SC Vet Center is not listed, even though it is closer than the Charlotte VC that is listed: 

Colombia, SC Vet Center       141-154 miles from MB VC  (122.22 ATCF)

## AC / Expected behavior
- [ ] Validate that order listed is linear distance 
- [ ] Verify distances against independents source (google maps?)
- [ ] Front end displays XX max number of facilities 
- [ ] Front end displays locations within 120 miles, sorting nearest to farthest, within max number 
- [ ] Locations must be able to be driven to from origin 

The assumption is that "Other nearby Vet Centers" should honor distance. 
How literally are we defining "nearby"? If this is not expected behavior, perhaps some qualifying text like "Not necessarily listed by distance"

@swirtSJW stated that distance calculated is a straight line (the logic does not compute driving distance)

## Screenshots
<img width="152" alt="image" src="https://user-images.githubusercontent.com/62269210/181386170-337f82ee-1f88-4bde-93bd-803fa8ba041f.png">

Google maps distance to Fayetteville VC:
<img width="647" alt="image" src="https://user-images.githubusercontent.com/62269210/181393058-66ee6c2c-acf1-4f4c-a443-b799d4573c48.png">

Google maps distance to Charleston VC 
<img width="640" alt="image" src="https://user-images.githubusercontent.com/62269210/181393174-479f47b3-da18-4b25-be3b-25e8ec4b8f36.png">

 ## Additional context
From what point is "nearby' calculated? The lat/long of the origin Vet Center? the center of the zip code? the center of the city? 
I know there is a bounding box vs radius issue, which might be contributing to this error?
I suspected this might be driving distance vs as-the-crow-flies distance, but my quick investigation (see screenshots) doesn't support that.

For reference, here is a VC Location List page in production: https://www.va.gov/west-los-angeles-vet-center/locations/

## Desktop (please complete the following information if relevant, or delete)
Issue is independent of OS/Browser

## Labels
(You can delete this section once it's complete)
- [x] Issue type (red) (defaults to "Defect")
- [ ] CMS subsystem (green)
- [ ] CMS practice area (blue)
- [x] CMS workstream (orange) (not needed for bug tickets)
- [ ] CMS-supported product (black)

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [ ] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [ ] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public Websites`
- [X] `⭐️ Facilities`
- [ ] `⭐️ User support`

**Comment by maxx1128** [comment]
For any devs having trouble reproducing the "Other Nearby Vety Centers" section locally, I got it working with these two changes:

1. At `NearByVetCenters.jsx:60`, replace `!props.togglesLoading` with `props.togglesLoading`
2. At `utilities/api/index.js:57`, replace `environment.API_URL` with `"https://api.va.gov/"`

**Comment by maxx1128** [comment]
I looked into different API options for calculating driving distance (not just as-the-crow-flies) and it turns out Mapbox already has an existing endpoint for doing this called [the Matrix endpoint.](https://docs.mapbox.com/api/navigation/matrix/#retrieve-a-matrix) It can calculate the distance and time for driving between two sets of coordinates. I tried it out for the Myrtle Beach example and it worked and was accurate for the distance and time showed on Google maps. It can also do multiple checks in a single request which is quicker and convenient.

My thought is this component does another request to this endpoint, get the distance and travel data, add it to the array of vet centers, and use it for filtering and sorting the list as needed. It would also be possible to include the time and distance estimates in the results themselves, but that will likely need feedback from design.
